### DISCUSSION TITLE
How to programmatically open and close Sheet?

### DISCUSSION THREAD

**Discussion by akubima** [huntabyte/shadcn-svelte#727] [Help] [answered]
How to programmatically open and close a Sheet? For example I use a Sheet as a sidebar in my project and I want the Sheet to close as soon as the user clicks on a link or when there is a change to the route.

Thanks.

**Comment by thedittmer** [comment]
I need this too. I'd like to close the sheet when the user click any link in the sheet, or when the page changes to a new route.

**Comment by thedittmer** [comment]
Eureka! I found the solution, the mobile-nav component.
https://github.com/huntabyte/shadcn-svelte/blob/1a3493aa8d27d074e0a2892d11cf8a919d0af3da/apps/www/src/lib/components/docs/nav/mobile-nav.svelte#L12

Create a variable called open and bind it to the sheet route. 
`<Sheet.Root bind:open>`

Import navigating from $app/stores, and set open to false when navigating.
`import { navigating } from '$app/stores';

	let open = false;
	$: if ($navigating) open = false;
`

**Reply by dhndmn** [reply]
Thanks! This helped me out. Just curious - what is special about the word 'open' as opposed to something custom like 'openSheet'. I was trying it the latter way and only when I changed it to simply 'open' did this start to work as intended.

**Reply by mstibbard** [reply]
`open` is from the [Bits UI Dialog primitive API](https://www.bits-ui.com/docs/components/dialog#dialogroot) which is what the Sheet component is built upon.

**Comment by robots4life** [comment, marked as answer]
```ts
let open = false;

<Sheet.Root bind:open>

<Sheet.Content side="left">

<a
	onclick={() => {
		open = false;
	}}
	href="/about"
	class="text-muted-foreground hover:text-foreground {$pageStore.route.id === '/about'
		? 'font-bold text-sky-950'
		: ''}">About</a
>
```
